### DISCUSSION TITLE
TS0121 Din smart relay (with power monitoring by polling)

### DISCUSSION THREAD

**Discussion by okastl** [Koenkk/zigbee2mqtt#13420] [Devices]
There is a new version of the TuYa Din smart relay, which requires polling similar to the newer version of the TuYa smart plug.
I have attached a working external converter for this device. Maybe someone (@Koenkk  ?) wants to add this to the herdsman repository.

[_TZ3000_aigddb2b.zip](https://github.com/Koenkk/zigbee2mqtt/files/9271019/_TZ3000_aigddb2b.zip)

**Comment by Koenkk** [comment]
Can you provide the data/database.db entry of this device?

**Reply by okastl** [reply]
> Can you provide the data/database.db entry of this device?

`{"id":6,"type":"Router","ieeeAddr":"0xcc86ecfffeeac236","nwkAddr":7328,"manufId":4098,"manufName":"_TZ3000_aigddb2b","powerSource":"Mains (single phase)","modelId":"TS0121","epList":[1],"endpoints":{"1":{"profId":260,"epId":1,"devId":81,"inClusterList":[0,4,5,6,1794,2820],"outClusterList":[25,10],"clusters":{"genBasic":{"attributes":{"65503":"\u0000\u0000\u0000\u0000\u0005","65506":30,"65534":0,"modelId":"TS0121","manufacturerName":"_TZ3000_aigddb2b","powerSource":1,"zclVersion":3,"appVersion":65,"stackVersion":0,"hwVersion":1,"dateCode":""}},"haElectricalMeasurement":{"attributes":{"acCurrentDivisor":1000,"acCurrentMultiplier":1,"rmsVoltage":229,"rmsCurrent":0,"activePower":0}},"seMetering":{"attributes":{"divisor":100,"multiplier":1,"currentSummDelivered":[0,0]}},"genOnOff":{"attributes":{"onOff":0,"onTime":0,"offWaitTime":0,"tuyaBacklightMode":1,"moesStartUpOnOff":2}}},"binds":[],"configuredReportings":[],"meta":{}}},"appVersion":65,"stackVersion":0,"hwVersion":1,"dateCode":"","zclVersion":3,"interviewCompleted":true,"meta":{"configured":1121114317},"lastSeen":1659782678638,"defaultSendRequestWhen":"immediate"}
`

**Reply by Koenkk** [reply]
How is this device detected in z2m 1.27.0? It should be detected as `TS0121_plug` which already uses polling

**Reply by okastl** [reply]
You are correct! I wasn't aware, that `TS0121_plug` is a "polling" variant. Somehow I had the impression, that it isn't working correctly.
